File: reverseLinkedList.c
#include <stdio.h>

struct Node {
    int data;
    struct Node* next;
};

struct Node* reverseList(struct Node* head){

    struct Node *curr = head;
    struct Node *prev = NULL;
    struct Node *next;

    while (curr != NULL){
        next = curr->next;
        curr->next = prev; //reverse current node's next pointer
        prev = curr;
        curr = next;
    }
    return prev;
}

struct Node* createNode(int new_data) {
    struct Node* new_node = (struct Node*)malloc(sizeof(struct Node));
    new_node->data = new_data;
    new_node->next = NULL;
    return new_node;
}

File: reattempt2/reverseLinkedList.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def reverseLinkedList(self, head: Optional[ListNode]) -> Optional[ListNode]:
            prev = None #points to NULL/None
            curr = head #points to head
            
            while(curr != None): #while curr hasn't reached to the end of linked list
                 next_node = curr.next #store next node
                 curr.next = prev #reverse current node's pointer
                 prev = curr #prev pointer to curr node
                 curr = next_node #curr pointer to next node

            return prev #where prev points is going to be the new head
                 